include/zenoh-pico/api/primitives.h

```diff
@@ -394,24 +394,52 @@ const char *zp_scouting_config_get(const z_loaned_scouting_config_t *config, uin
 int8_t zp_scouting_config_insert(z_loaned_scouting_config_t *config, uint8_t key, const char *value);
 
 /**
- * Constructs a :c:type:`z_encoding_t`.
+ * Constructs a :c:type:`z_owned_encoding_t`.
  *
  * Parameters:
- *   prefix: A known :c:type:`z_encoding_prefix_t`.
- *   suffix: A custom suffix to be appended to the prefix.
+ *   encoding: a reference to an uninitialized :c:type:`z_owned_encoding_t`
+ *   id: A known :c:type:`z_encoding_id_t`.
+ *   schema: A custom schema string value.
  *
  * Returns:
- *   Returns the constructed :c:type:`z_encoding_t`.
+ *   Returns ``0`` if construction is successful, or a ``negative value`` otherwise.
  */
-z_encoding_t z_encoding(z_encoding_prefix_t prefix, const char *suffix);
+int8_t zp_encoding_make(z_owned_encoding_t *encoding, z_encoding_id_t id, const char *schema);
 
 /**
- * Constructs a default encoding.
+ * Constructs a :c:type:`z_owned_encoding_t` with default value.
+ *
+ * Parameters:
+ *   encoding: a reference to an uninitialized :c:type:`z_owned_encoding_t`
  *
  * Returns:
- *   Returns the constructed :c:type:`z_encoding_t`.
+ *   Returns ``0`` if construction is successful, or a ``negative value`` otherwise.
+ */
+int8_t zp_encoding_default(z_owned_encoding_t *encoding);
+
+/**
+ * Returns ``true`` if encoding is in non-default state, ``false`` otherwise.
  */
-z_encoding_t z_encoding_default(void);
+_Bool z_encoding_loan_check(const z_loaned_encoding_t *encoding);
```

Comment:
When I realized encoding was never truly tested in the example, I tried to add some code and came up with this on the callback of a subscriber ( }not pushed on this PR):
```C
    const z_loaned_encoding_t *encoding = z_sample_encoding(sample);
    if (z_check(encoding)) {
        printf("     with encoding {id: %d, schema: '%.*s'}\n", encoding->id, (int)encoding->schema.len,
               encoding->schema.start);
```
Not sure it's useful but it could be.

Reply:
But it seems that if we don't have encodings, z_sample_encoding should return NULL. No?